lib/sycamore/sycamore/evaluation/evaluate.py

```diff
@@ -0,0 +1,139 @@
+        datapoint.question = template_Q.format(datapoint.raw["Question"], datapoint.filters.get(filter_Q))
+    return datapoint
+
+
+class Assessment(ABC):
+    @abstractmethod
+    def run_evaluation(self, ctx: Context, index: str, **kwargs):
+        pass
+
+    def __call__(self, context: Context, index: str, **kwargs):
+        return self.run_evaluation(context, index, **kwargs)
+
+
+class QualityAssessment(Assessment):
+    def __init__(self, GT_path: str, rag_config: Dict, **kwargs):
+        self.user = os.environ.get("USER", os.environ.get("USERNAME"))
+        self.gt_path = GT_path
+        self.rag_config = rag_config
+        self.os_client_args = kwargs.get("os_client_args", "")
+        self.metrics = kwargs.get("metrics", [document_retrieval_metrics, rouge_metrics])
+        self.custom_question_augmentation = kwargs.get("custom_question_augmentation", {})
+        self.question_augmentation_filter = kwargs.get("question_augmentation_filter", "")
+
+    @staticmethod
+    def create_evaluation_datapoint(
+        json_dict: Dict, custom_question_augmentation: str = "{}", question_augmentation_filter: str = ""
+    ):
+        result = []
+        assert json_dict is not None
+        assert isinstance(json_dict, dict)
+        for datapoint in json_dict["data"][:]:
+            document = EvaluationDataPoint()
+            datapoint["custom_question_augmentation"] = custom_question_augmentation
+            datapoint["question_augmentation_filter"] = question_augmentation_filter
+            document.raw = datapoint
+            document.ground_truth_answer = datapoint["Answer"]
+            document.filters = datapoint.get("Filters", {})
+            result += [{"doc": document.serialize()}]
+        return result
+
+    def run_evaluation(self, ctx: Context, index: str, **kwargs):
+        custom_question_augmentation = str(self.custom_question_augmentation)
+        question_augmentation_filter = str(self.question_augmentation_filter)
+        input_docset = ctx.read.json(
+            paths=self.gt_path,
+            doc_extractor=lambda json_dict: QualityAssessment.create_evaluation_datapoint(
+                json_dict, custom_question_augmentation, question_augmentation_filter
+            ),
+        ).map(add_filters_to_question)
+
+        pipeline = EvaluationPipeline(
+            index=index, os_client_args=self.os_client_args, os_config=self.rag_config, metrics=self.metrics
+        )
+        query_level_metrics, aggregated_metrics = pipeline.execute(input_docset)
+        return query_level_metrics.take_all(), aggregated_metrics
+
+
+class Evaluate:
+    """
+    The Evaluate runs the evaluation test on
+    Index or list of indices against a ground truth
+
+    Args:
+        context: The Sycamore context to use
+        index: Index or list of Index
+        assessment: The Assessment to run
+        GT_path: The path to ground truth
```

Comment:
Why are these things listed on the `Evaluate` class docstring rather than `QualityAssessment`?

Reply:
This is a mistake,  I will move it to relevant class